planner/core/planbuilder.go

```diff
@@ -1004,8 +1004,20 @@ func getLatestIndexInfo(ctx sessionctx.Context, id int64, startVer int64) (map[i
 		return nil, false, nil
 	}
 	latestIndexes := make(map[int64]*model.IndexInfo)
-	latestTbl, exist := is.TableByID(id)
-	if exist {
+
+	var latestTbl table.Table
+	latestTblExist := false
+
+	localTemporaryTables := ctx.GetSessionVars().LocalTemporaryTables
+	if localTemporaryTables != nil {
+		latestTbl, latestTblExist = localTemporaryTables.(*infoschema.LocalTemporaryTables).TableByID(id)
+	}
+
+	if !latestTblExist {
+		latestTbl, latestTblExist = is.TableByID(id)
+	}
+
```

Comment:
Confused here.
In my mind. the temporary table can only survive in the session. That's means other session's ddl can't see the temporary meta and data, which couldn't cause schema-changed in your current session transaction.

```
other session ddl ----------------------ddl--------------------------------->
                                         | (can't see temporary table's meta?)
                                         v

begin(get snapshot)  ------(dml perceive the latest schema)-------->    commit
```

Reply:
The modification here is not to handle temporary table's schema change. In fact, we do not even allow the alter operation for temporary table. This problem is that here used a infoschema got from domain and didn't include the temporary table(because local temporary table's schema is stored in memory, not in tikv). So to get the right table schema, we must  perform another get from session. LocalTemporaryTables

Reply:
Got it. impressive